server/routes/map.js

```diff
@@ -115,7 +115,15 @@ router.get("/query", async function (req, res, next) {
             };
         });
 
-        results = results.filter(place => place.duration.value <= timeLimit);
+        results = results.filter(place =>
+        {
+            if (place.duration === undefined)
```

Comment:
You could do this in one line with
`return place && place.duration && place.duration.value <= timeLimit`